Src/Newtonsoft.Json/JsonTextReader.Async.cs

```diff
@@ -0,0 +1,2327 @@
+                            {
+                                finished = true;
+                            }
+                        }
+                        else
+                        {
+                            _charPos++;
+                        }
+                        break;
+                    case StringUtils.CarriageReturn:
+                        await ProcessCarriageReturnAsync(false, cancellationToken).ConfigureAwait(false);
+                        break;
+                    case StringUtils.LineFeed:
+                        ProcessLineFeed();
+                        break;
+                    default:
+                        if (currentChar == ' ' || char.IsWhiteSpace(currentChar))
+                        {
+                            ateWhitespace = true;
+                            _charPos++;
+                        }
+                        else
+                        {
+                            finished = true;
+                        }
+                        break;
+                }
+            }
+
+            return !oneOrMore || ateWhitespace;
+        }
+
+        private async Task ParseStringAsync(char quote, ReadType readType, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            _charPos++;
+
+            ShiftBufferIfNeeded();
+            await ReadStringIntoBufferAsync(quote, cancellationToken).ConfigureAwait(false);
+            SetPostValueState(true);
+
+            switch (readType)
+            {
+                case ReadType.ReadAsBytes:
+                    Guid g;
+                    byte[] data;
+                    if (_stringReference.Length == 0)
+                    {
+                        data = new byte[0];
+                    }
+                    else if (_stringReference.Length == 36 && ConvertUtils.TryConvertGuid(_stringReference.ToString(), out g))
+                    {
+                        data = g.ToByteArray();
+                    }
+                    else
+                    {
+                        data = Convert.FromBase64CharArray(_stringReference.Chars, _stringReference.StartIndex, _stringReference.Length);
+                    }
+
+                    SetToken(JsonToken.Bytes, data, false);
+                    break;
+                case ReadType.ReadAsString:
+                    string text = _stringReference.ToString();
+
+                    SetToken(JsonToken.String, text, false);
+                    _quoteChar = quote;
+                    break;
+                case ReadType.ReadAsInt32:
+                case ReadType.ReadAsDecimal:
+                case ReadType.ReadAsBoolean:
+
+                    // caller will convert result
+                    break;
+                default:
+                    if (_dateParseHandling != DateParseHandling.None)
```

Comment:
I don't see any awaits inside this default. Can its code be moved to a shared method?